sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -117,9 +117,14 @@ object ExpressionEncoder {
         s"`GetColumnByOrdinal`, but there are ${getColExprs.size}")
 
       val input = GetStructField(newDeserializerInput, index)
-      enc.objDeserializer.transformUp {
+      val newDeserializer = enc.objDeserializer.transformUp {
         case GetColumnByOrdinal(0, _) => input
       }
+      if (enc.objSerializer.nullable) {
+        If(IsNull(input), Literal.create(null, newDeserializer.dataType), newDeserializer)
```

Comment:
Can we create a new JIRA ticket and PR for this bug?

Reply:
👋  I've already created [SPARK-37829](https://issues.apache.org/jira/browse/SPARK-37829). What do you mean exactly?